libraries/chain/block_log.cpp

```diff
@@ -87,20 +87,25 @@ namespace eosio { namespace chain {
 
       template <typename Stream>
       void write_to(Stream& ds) const {
-         EOS_ASSERT(version >= block_x_start_version, block_log_exception, "this method does not support writeing block log ${version}",
-                    ("version", version));
          ds.write(reinterpret_cast<const char*>(&version), sizeof(version));
-         ds.write(reinterpret_cast<const char*>(&first_block_num), sizeof(first_block_num));
+         if (version != initial_version) {
+            ds.write(reinterpret_cast<const char*>(&first_block_num), sizeof(first_block_num));
 
-         std::visit(overloaded{[&ds](const chain_id_type& id) { ds << id; },
-                               [&ds](const genesis_state& state) {
-                                  auto data = fc::raw::pack(state);
-                                  ds.write(data.data(), data.size());
-                               }},
-                    chain_context);
+            std::visit(overloaded{[&ds](const chain_id_type& id) { ds << id; },
+                                 [&ds](const genesis_state& state) {
+                                    auto data = fc::raw::pack(state);
+                                    ds.write(data.data(), data.size());
+                                 }},
+                     chain_context);
 
-         auto totem = block_log::npos;
-         ds.write(reinterpret_cast<const char*>(&totem), sizeof(totem));
+            auto totem = block_log::npos;
+            ds.write(reinterpret_cast<const char*>(&totem), sizeof(totem));
+         }
+         else {
+            const auto& state = std::get<genesis_state>(chain_context);
```

Comment:
(Missed this) If we are opening this up to supporting initial_version, then we need to add some logic to reset to change the version so that the header information remains consistent to keep the block log valid, unless we can see that reset is not allowed to be called with version == 1.

Reply:
This only happens if reset is called after calling set_default, which is a test method, so it won't.